opentech/apply/funds/models.py

```diff
@@ -482,44 +486,36 @@ def save(self, *args, **kwargs):
                 self.workflow_name = self.page.workflow_name
             self.status = str(self.workflow.first())
 
+        # add a denormed version of the answer for searching
+        self.search_data = ' '.join(self.prepare_search_values())
+
         return super().save(*args, **kwargs)
 
-    def render_answers(self):
-        context = {'fields': list()}  # type: ignore
-        for field in self.form_fields:
+    def data_and_fields(self):
+        for stream_value in self.form_fields:
             try:
-                data = self.form_data[field.id]
+                data = self.form_data[stream_value.id]
             except KeyError:
                 pass  # It was a named field or a paragraph
             else:
-                form_field = field.block.get_field(field.value)
-                data = self.prepare_value(form_field, data)
-                context['fields'].append({
-                    'field': form_field,
-                    'value': data,
-                })
-        return render_to_string(self.field_template, context)
-
-    def prepare_value(self, field, data):
-        NO_RESPONSE = 'No response'
-        if hasattr(field, 'choices'):
-            if not data:
-                return [NO_RESPONSE]
-
-            if isinstance(data, str):
-                data = [data]
-            choices = dict(field.choices)
-            try:
-                data = [choices[value] for value in data]
-            except KeyError:
-                data = [choices[int(value)] for value in data if value]
-        else:
-            if not data and not isinstance(data, bool):
-                return NO_RESPONSE
+                yield data, stream_value
 
-            data = str(data)
-
-        return data
+    def render_answers(self):
+        fields = [
+            field.render(context={'data': data})
+            for data, field in self.data_and_fields()
+        ]
+        return mark_safe(''.join(fields))
+
+    def prepare_search_values(self):
+        excluded_fields = ['data', 'time', 'datetime', 'value', 'category', 'number']
+        for data, stream in self.data_and_fields():
+            if stream.block_type not in excluded_fields:
+                value = stream.block.get_searchable_content(stream.value, data)
```

Comment:
My gut reaction here was "we should handle the edge case where the block doesn't implement this", but then again it is defined on FormFieldBlock, so that's ok